src/Sylius/Bundle/LocaleBundle/Templating/Helper/LocaleHelper.php

```diff
@@ -41,6 +42,188 @@ public function getCurrentLocale()
     }
 
     /**
+     * @return string
+     */
+    public function getCurrentCalendar()
+    {
+        return $this->localeContext->getCurrentCalendar();
+    }
+
+    /**
+     * @return string
+     */
+    public function getCurrentDirection()
+    {
+        return $this->localeContext->getCurrentDirection();
+    }
+
+    /**
+     * Format date in a calendar-aware fashion
+     *
+     * @param string|\DateTime     $date     Date to format
+     * @param string               $format   PHP-compatible date format
+     * @param string|\DateTimeZone $timezone Timezone
+     * @param string               $calendar Calendar name
+     *
+     * @return bool|string Calendar-aware localized formatted date
+     */
+    public function formatDate($date, $format = null, $timezone = null, $locale = null, $calendar = null)
+    {
+        // Determine the timezone
+        if (!$timezone) {
+            $timezone = date_default_timezone_get();
+        }
+
+        if (!$timezone instanceof \DateTimeZone) {
+            $timezone = new \DateTimeZone($timezone);
+        }
+
+        if ($locale === null) {
+            $locale = $this->localeContext->getCurrentLocale();
+        }
+
+        if ($calendar === null) {
+            $calendar = $this->localeContext->getCurrentCalendar();
+        }
+
+        if ($date instanceof \DateTime || $date instanceof \DateTimeInterface) {
+            $date = clone $date;
+
+            if (false !== $timezone) {
+                $date->setTimezone($timezone);
+            }
+        } else {
+            // If $date's string representation is somehow numeric,
+            // we'll assume it's a timestamp then prepend an @ and pass it to \DateTime.
+            $asString = (string) $date;
+            if (ctype_digit($asString) || (!empty($asString) && '-' === $asString[0] && ctype_digit(substr($asString, 1)))) {
+                $date = '@' . $date;
+            }
+
+            $date = new \DateTime($date, $timezone);
+        }
+
+        if ('gregorian' === $calendar || null === $calendar) {
+            return $date->format($format);
+        } else {
+            $dateFormatter = new \IntlDateFormatter("$locale@calendar=$calendar", null, null, $timezone, \IntlDateFormatter::TRADITIONAL, self::convertDatePhpToIcu($format));
```

Comment:
Unfortunately it [only supports English!](https://github.com/symfony/symfony/blob/2.7/src/Symfony/Component/Intl/DateFormatter/IntlDateFormatter.php#L132)